homeassistant/config.py

```diff
@@ -1015,21 +1029,115 @@ async def merge_packages_config(
     return config
 
 
+@callback
+def async_process_component_config_errors(
+    hass: HomeAssistant,
+    integration: Integration,
+    config_exception_info: list[ConfigExceptionInfo],
+    raise_on_failure: bool = False,
+) -> None:
+    """Process component configuration errors."""
+
+    p_ex: ConfigExceptionInfo
+    config_error_messages: list[
+        tuple[str, ConfigExceptionInfo, str, str, int | str]
+    ] = []
+    general_error_messages: list[tuple[str, ConfigExceptionInfo]] = []
+    domain = integration.domain
+    for p_ex in config_exception_info:
+        link = p_ex.integration_link or integration.documentation
+        ex = p_ex.ex
+        p_name = p_ex.platform_name
+        p_config = p_ex.config
+        config_file: str = "?"
+        line: int | str = "?"
```

Comment:
Why do we default these to "?", just use the return value from `find_annotation` instead: `tuple[str, int | str] | None`
```suggestion
        annotation: tuple[str, int | str] | None = None
```

Reply:
Yes, so? My point is that we don't need to resolve to a string here. Attach the output of `find_annotation` to `config_error_messages` instead.